File: answers/lintcode/Valid Palindrome.cpp
class Solution {
public:
    /**
     * @param s A string
     * @return Whether the string is a valid palindrome
     */
    bool isPalindrome(string& s) {
        // Write your code here
		if (0 == s.length()) {
			return true;
		}
		size_t left = 0;
		size_t right = s.length() - 1;
		while (left < right && right < s.length()) {
			if (isalnum(s[left]) && isalnum(s[right])) {
				if (tolower(s[left]) != tolower(s[right])) {
					return false;
				}
				else {
					++ left;
					-- right;
				}
			}
			if (! isalnum(s[left])) {
				++ left;
			}
			if (! isalnum(s[right])) {
				-- right;
			}
		}
		return true;
    }
};

File: answers/other/in sap/Valid Palindrome.py
class Solution(object):
	def isPalindrome(self, s):
		"""
		:type s: str
		:rtype: bool
		"""
		
		left = 0
		right = len(s) - 1
		while left < right:
			left_flag = s[left].isalnum()
			right_flag = s[right].isalnum()
			if left_flag and right_flag:
				if s[left].lower() != s[right].lower():
					return False
				left += 1
				right -= 1
			if not left_flag:
				left += 1
			if not right_flag:
				right -= 1
		return True